sycl/source/detail/memory_manager.cpp

```diff
@@ -255,34 +289,44 @@ void copyH2D(SYCLMemObjI *SYCLMemObj, char *SrcMem, QueueImplPtr,
   assert(SYCLMemObj && "The SYCLMemObj is nullptr");
 
   const RT::PiQueue Queue = TgtQueue->getHandleRef();
-  // Adjust first dimension of copy range and offset as OpenCL expects size in
-  // bytes.
-  DstSize[0] *= DstElemSize;
   const detail::plugin &Plugin = TgtQueue->getPlugin();
-  if (SYCLMemObj->getType() == detail::SYCLMemObjI::MemObjType::BUFFER) {
-    DstOffset[0] *= DstElemSize;
-    SrcOffset[0] *= SrcElemSize;
-    SrcAccessRange[0] *= SrcElemSize;
-    DstAccessRange[0] *= DstElemSize;
-    SrcSize[0] *= SrcElemSize;
 
+  detail::SYCLMemObjI::MemObjType MemType = SYCLMemObj->getType();
+  term_positions SrcPos, DstPos;
+  prepTermPositions(SrcPos, DimSrc, MemType);
+  prepTermPositions(DstPos, DimDst, MemType);
+
+  //  If the Dimension is 1, and being called by ~SYCLMemObjT,
+  //  thenDstAccessRange[0] and DstSize[0] will already sized to bytes with
+  //  DstElemSize of 1.
+  size_t DstXOffBytes = DstOffset[DstPos.x_term] * DstElemSize;
+  size_t SrcXOffBytes = SrcOffset[SrcPos.x_term] * SrcElemSize;
+  size_t DstARWidthBytes = DstAccessRange[DstPos.x_term] * DstElemSize;
```

Comment:
What "AR" stands for in the variable name?

Reply:
AccessRange